pkg/utils/httpserverutils.go

```diff
@@ -412,18 +412,32 @@ func NewSingleESResponse() *SingleESResponse {
 
 func WriteResponse(ctx *fasthttp.RequestCtx, httpResp HttpServerResponse) {
 	ctx.SetContentType(ContentJson)
-	jval, _ := json.Marshal(httpResp)
-	_, err := ctx.Write(jval)
+	jval, err := json.Marshal(httpResp)
 	if err != nil {
+		log.Errorf("WriteResponse: Error marshaling HTTP response: %v", err)
```

Comment:
nit: I'm not sure the `err` will include enough info about the `httpResp` to debug what went wrong. If it doesn't, you should include `httpResp` in the log message.

Similar comment for lines 423, 433, 439